tc/sgx/trusted_worker_manager/enclave/CMakeWorkloads.txt

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##################################################################
+MACRO(CREATE_SUPPORTED_WORKLOADS_LIST)
+    SET(SUPPORTED_WORKLOADS_LIST "")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "echo-result")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "heart-disease-eval")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "inside-out-eval")
+    LIST(APPEND SUPPORTED_WORKLOADS_LIST "simple-wallet")
+
+    # A new workload can be added by using below syntax
+    # LIST(SUPPORTED_WORKLOADS_LIST "<workload_id>")
+ENDMACRO()
+
+##################################################################
+MACRO(CREATE_SUPPORTED_WORKLOAD_LIBRARY_LIST)
+    SET(SUPPORTED_WORKLOAD_LIBRARY_LIST "")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "echo_wo")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "heart_disease_eval")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "inside_out_eval")
+    LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "simple_wallet")
+
+    # A new workload library can be added by using below syntax
+    # Note: Order of workload id should match the order in
+    #       supported workload library list
+    # LIST(APPEND SUPPORTED_WORKLOAD_LIBRARY_LIST "<workload_lib_name>")
+ENDMACRO()
+
+##################################################################
+MACRO(CREATE_LINK_WORKLOAD_LIBRARY_LIST)
+    SET(WORKLOAD_LIST "$ENV{WORKLOADS}")
+    SET(WORKLOAD_LIB_LIST "")
+    # Create workload library list based on workloads passed
+    IF("${WORKLOAD_LIST}" STREQUAL "all" OR "${WORKLOAD_LIST} " STREQUAL " ")
+        SET(WORKLOAD_LIB_LIST ${SUPPORTED_WORKLOAD_LIBRARY_LIST})
+    ELSE()
+        FOREACH(workload ${WORKLOAD_LIST})
+            # Check if user provided workload is present in supported workload list
+            # If not found, return -1
+            LIST(FIND SUPPORTED_WORKLOADS_LIST ${workload} index)
+            IF(NOT ${index} EQUAL -1)
+                # Get workload library corresponding to workload identifier
+                LIST(GET SUPPORTED_WORKLOAD_LIBRARY_LIST ${index} workload_lib)
+                LIST(APPEND WORKLOAD_LIB_LIST ${workload_lib})
+            ELSE()
+                MESSAGE(FATAL_ERROR,
+                    "Unable to find workload library corresponding to parsed workload")
+            ENDIF()
+        ENDFOREACH(workload)
+    ENDIF()
+ENDMACRO()
+
+##################################################################
+MACRO(CONFIG_WORKLOAD_LINKER_FLAGS)
```

Comment:
CMAKE_SHARED_LINKER_FLAGS for workloads should be set only for workload ids passed by the user. To make things more simple, we could use the build sub folder name same as workload library name. "build/<workload_lib>". Then you can iterate over the WORKLOAD_LIB_LIST.

Reply:
Renamed static library name inside echo workload cmake file to match the folder name and inside_out workload folder name to match the library name. Rest of the example workloads folder name and static library names are already same.
With all these changes, now i iterated over workload_lib_list to link cmake linker flags to locate library paths.